ADAL/src/ui/mac/ADAuthenticationViewController.m

```diff
@@ -0,0 +1,212 @@
+// Copyright © Microsoft Open Technologies, Inc.
+//
+// All Rights Reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+// ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+// PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache License, Version 2.0 for the specific language
+// governing permissions and limitations under the License.
+
+#import "ADWebAuthDelegate.h"
+#import "ADAuthenticationViewController.h"
+#import "ADLogger.h"
+#import "ADALFrameworkUtils.h"
+
+#define DEFAULT_WINDOW_WIDTH 420
+#define DEFAULT_WINDOW_HEIGHT 650
+
+static CGRect _CenterRect(CGRect rect1, CGRect rect2)
+{
+    CGFloat x = rect1.origin.x + ((rect1.size.width - rect2.size.width) / 2);
+    CGFloat y = rect1.origin.y + ((rect1.size.height - rect2.size.height) / 2);
+    
+    x = x < 0 ? 0 : x;
```

Comment:
Collapsing all of this code to a single line would make things considerably more difficult to read and wouldn't reduce the number of virtual variables in the AST anyways.